aten/src/ATen/native/sparse/SparseMatMul.cpp

```diff
@@ -191,25 +207,18 @@ void sparse_matmul_kernel(
   */
 
   auto M = mat1.size(0);
-  auto K = mat1.size(1);
   auto N = mat2.size(1);
 
-  auto mat1_indices_ = mat1._indices().contiguous();
-  auto mat1_values = mat1._values().contiguous();
-  Tensor mat1_row_indices = mat1_indices_.select(0, 0);
-  Tensor mat1_col_indices = mat1_indices_.select(0, 1);
-
-  Tensor mat1_indptr = coo_to_csr(mat1_row_indices.data_ptr<int64_t>(), M, mat1._nnz());
-
-  auto mat2_indices_ = mat2._indices().contiguous();
-  auto mat2_values = mat2._values().contiguous();
-  Tensor mat2_row_indices = mat2_indices_.select(0, 0);
-  Tensor mat2_col_indices = mat2_indices_.select(0, 1);
-
-  Tensor mat2_indptr = coo_to_csr(mat2_row_indices.data_ptr<int64_t>(), K, mat2._nnz());
+  const auto mat1_csr = mat1.to_sparse_csr();
+  const auto mat2_csr = mat2.to_sparse_csr();
```

Comment:
Start with
```suggestion
  const auto mat2_csc = mat2.to_sparse_csc();
```
for a much faster kernel.

Reply:
Faster in theory, but not yet in practice. Please check the comment above.

Reply:
Resolving as OT within this PR.